velox/exec/fuzzer/FuzzerUtil.cpp

```diff
@@ -36,6 +52,37 @@ void writeToFile(
   writer.close();
 }
 
+// Recursive function to create splits with their corresponding schemas and
+// store in splits.
+// In a table directory, each partition would be stored as a
+// sub-directory, multiple partition columns would make up nested directory
+// structure.
+//
+// For example for a file path such as /p0=0/p1=0/0000_file1, creates
+// split with partition keys (p0, 0), (p1 0)
+void makeSplitsWithSchema(
+    std::vector<Split>& splits,
+    const std::string& directory,
+    std::unordered_map<std::string, std::optional<std::string>> partitionKeys) {
+  for (auto const& entry : std::filesystem::directory_iterator{directory}) {
+    if (entry.is_directory()) {
+      auto directoryName = entry.path().string();
+      auto partition =
+          extractPartition(directoryName.substr(directory.size() + 1));
+      partitionKeys.insert(
+          {partition.first,
+           partition.second == DEFAULT_PARTITION_VALUE
```

Comment:
in which case  the partition value is null?

Reply:
```
CREATE TABLE test_k (
    name VARCHAR,
    city VARCHAR
)
WITH (
    partitioned_by = ARRAY['city'],
    retention_days = 30,
    uii = FALSE
)

insert into test_k values ('hello', NULL),('day', 'WUHAN')
```

Reply:
I see. where is DEFAULT_PARTITION_VALUE defined?